### PR TITLE
GEODE-6998: Add index test for gii tombstone/values update to tombstone 

### PR DESCRIPTION
**Pull request by jhuynh1**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jhuynh1** [comment]
@mivanac Here is the dunit test that reproduced the issue and is now fixed thanks to your commit

**Comment by mivanac** [comment]
Thanks

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/PersistentRegionCompactRangeIndexDUnitTest.java [hunk 1]
```diff
@@ -151,6 +151,50 @@ public void giiWithPersistenceAndStaleDataDueToUpdatesShouldCorrectlyPopulateInd
     });
   }
 
+  @Test
+  public void giiWhereNewAndOldValueIsTombstoneShouldNotThrowNPE()
+      throws Exception {
+    // This test requires the oldKeyValuePair to be instantiated, to do so, we need the gii thread
+    // to gii some updates or removes on existing (non-tombstone) values before processing the
+    // scenario of tombstone to tombstone
+    String regionName = "replicateTestRegion";
```
Comment (nabarunnag) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/PersistentRegionCompactRangeIndexDUnitTest.java]: Looks fine to me. I was just wondering about the naming in our tests. This region name has "replicated" in it but the other tests have regions names like "persistentTestRegionWithEntrySetIndex" which does not tell us what region type it is unless we view the properties file.

File: geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/PersistentRegionCompactRangeIndexDUnitTest.java [hunk 2]
```diff
@@ -151,6 +151,50 @@ public void giiWithPersistenceAndStaleDataDueToUpdatesShouldCorrectlyPopulateInd
...
+    int numEntries = 10;
+    Map<String, Portfolio> entries = new HashMap<>();
+    IntStream.range(0, numEntries).forEach(i -> entries.put("key-" + i, new Portfolio(i)));
+    HashMap halfEntries = new HashMap();
```
Comment (nabarunnag) [inline geode-core/src/distributedTest/java/org/apache/geode/cache/query/dunit/PersistentRegionCompactRangeIndexDUnitTest.java]: Can this be a set instead?

**Comment by jhuynh1** [comment]
The failing dunit test has since been fixed on develop and was not due to this diff.

**Pull request merged**